blocks/bricks/recurrent.py

```diff
@@ -772,10 +773,10 @@ def suffixes(names, level):
     @staticmethod
     def split_suffix(name):
         # Target name with suffix to the correct layer
-        name_level = name.split('_')
-        if len(name_level) == 2:
-            name, level = name_level
-            level = int(level)
+        name_level = name.split(RECURRENTSTACK_SEPARATOR)
+        if name_level[-1].isdigit():
```

Comment:
What if a stack element has a state with a name ending with a digit? If would rather change this condition to `if len(name_level) == 2`. You can also check that `len(name_level) < 3` and raise `ValueError` if it is not the case.